src/proxy/IMap.ts

```diff
@@ -269,6 +269,13 @@ export interface IMap<K, V> extends DistributedObject {
      */
     remove(key: K, value?: V): Promise<V | boolean>;
 
+    /**
+     * Removes all entries which match with the supplied predicate.
```

Comment:
You can add the following sentence as well:

- If this map has index, matching entries will be found via index search, otherwise they will be found by full-scan.